parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -110,6 +131,11 @@ public void writeTo(OutputStream out) throws IOException {
     }
 
     @Override
+    public byte[] getBytesUnsafe() {
+      return getBytes();
```

Comment:
As different implementations of Binary are used via Binary interface, exposing start and length values will require Binary interface to be aware of it. Correct me if I am not getting your point.